src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -60,27 +61,49 @@ public function process(ContainerBuilder $container)
 
     private function registerHandlers(ContainerBuilder $container)
     {
+        $definitions = array();
         $handlersByMessage = array();
 
         foreach ($container->findTaggedServiceIds($this->handlerTag, true) as $serviceId => $tags) {
             foreach ($tags as $tag) {
-                $handles = isset($tag['handles']) ? array($tag['handles']) : $this->guessHandledClasses($r = $container->getReflectionClass($container->getDefinition($serviceId)->getClass()), $serviceId);
+                if (isset($tag['handles'])) {
+                    $handles = isset($tag['method']) ? array($tag['handles'] => $tag['method']) : array($tag['handles']);
+                } else {
+                    $handles = $this->guessHandledClasses($r = $container->getReflectionClass($container->getDefinition($serviceId)->getClass()), $serviceId);
+                }
+
                 $priority = $tag['priority'] ?? 0;
 
-                foreach ($handles as $messageClass) {
+                foreach ($handles as $messageClass => $method) {
+                    if (is_int($messageClass)) {
```

Comment:
let's add a `\` in front of `is_*` calls to be consistent with the existing one above